### PR TITLE
refactor: separate search ui logic from frontend

### PR DESCRIPTION
**Pull request by Junjiequan**
## Description

This PR aims to separate the implementation of default user settings logic from FE.

## Motivation

Reduce maintenance costs for the future.

## Changes:

- Updated `frontend.config` to include new labelMaps and reorganize filter settings.
- Renamed and moved localColumns from `frontend.config`  into defaultDatasetsListSettings field.
- Removed unused interceptors for user settings.
- Added validation for filter components in findByIdUserSettings to check for valid filters based on FilterComponentType, and reset if invalid filters are found during validation.
- Adjusted findOneAndUpdateUserSettings to ensure proper upsert and default behavior

## Tests included

- [ ] Included for each change/fix?
- [ ] Passing? (Merge will not be approved unless this is checked) 

## Documentation
- [ ] swagger documentation updated \[required\]
- [ ] official documentation updated \[nice-to-have]

### official documentation info
If you have updated the official documentation, please provide PR # and URL of the pages where the updates are included

### CODE REVIEW

File: src/users/schemas/user-settings.schema.ts [hunk 1]
```diff
@@ -74,12 +71,12 @@ export class UserSettings {
...
-    default: FILTER_CONFIGS,
+    default: [],
     description: "Array of filters the user has set",
   })
   @Prop({
     type: [{ type: Object }],
-    default: FILTER_CONFIGS,
+    default: [],
   })
   filters: FilterConfig[];
```
Comment (Junjiequan) [inline src/users/schemas/user-settings.schema.ts]: FilterConfig needs to be changed.
It should be similar to:
```suggestion
export type FilterComponentType =
  | "LocationFilterComponent"
  | "PidFilterComponent"
  | "PidFilterContainsComponent"
  | "PidFilterStartsWithComponent"
  | "GroupFilterComponent"
  | "TypeFilterComponent"
  | "KeywordFilterComponent"
  | "DateRangeFilterComponent"
  | "TextFilterComponent";

// Define the Filter interface
export interface FilterConfig {
  FilterComponentType: string;
  visible: boolean;
}
```

File: src/users/users.controller.ts [hunk 2]
```diff
@@ -100,6 +94,7 @@ export class UsersController {
 
   @ApiBody({ type: CredentialsDto })
   @AllowAny()
+  @UseInterceptors(UserSettingsInterceptor)
```
Comment (nitrosx) [inline src/users/users.controller.ts]: I do not think we need this interceptor on login endpoint.

File: src/users/users.service.ts [hunk 3]
```diff
@@ -273,16 +274,27 @@ export class UsersService implements OnModuleInit {
   }
 
   async findByIdUserSettings(userId: string): Promise<UserSettings | null> {
```
Comment (nitrosx) [inline src/users/users.service.ts:275]: _findUserSettingsById_ is a better name

File: src/users/interceptors/user-settings.interceptor.ts
Comment (nitrosx) [inline src/users/interceptors/user-settings.interceptor.ts:1]: Do we need this interceptor?

**Review by nitrosx — CHANGES_REQUESTED** [review]

File: src/users/users.service.ts (hunk 3 above)
Comment (nitrosx) [reply src/users/users.service.ts:275]: The renaming of the internal functions needs more brainstorming.
It requires more effort in order to keep naming consistent.

File: src/users/dto/update-user-settings.dto.ts [hunk 4]
```diff
@@ -18,13 +10,14 @@ export class UpdateUserSettingsDto {
...
-  @IsArray()
-  readonly conditions: ScientificCondition[];
+  @ApiProperty({
+    type: "object",
+    additionalProperties: { type: "array", items: {} },
+    required: false,
+  })
+  @IsOptional()
+  @IsObject()
+  readonly frontendSettings?: Record<string, unknown[]>;
```
Comment (nitrosx) [inline src/users/dto/update-user-settings.dto.ts]: On second thought, can we call them just `extraSettings`?

**Review by nitrosx — CHANGES_REQUESTED** [review]
Minor changes.

File: src/users/dto/update-user-settings.dto.ts (hunk 4 above)
Comment (Junjiequan) [reply src/users/dto/update-user-settings.dto.ts:20]: extraSettings are too vague. I strongly against naming it as extraSettings.

File: src/users/users.controller.ts (hunk 2 above)
Comment (Junjiequan) [reply src/users/users.controller.ts]: It's removed in the following commit

File: src/users/dto/update-user-settings.dto.ts (hunk 4 above)
Comment (Junjiequan) [reply src/users/dto/update-user-settings.dto.ts]: ExtraSettings seems a bit too vague. I would kindly suggest reconsidering the name to something more specific.

File: src/users/interceptors/user-settings.interceptor.ts
Comment (Junjiequan) [reply src/users/interceptors/user-settings.interceptor.ts:1]: It's removed in the following commit.

**Review by nitrosx — APPROVED** [review]

**Pull request merged**